cmd/snapd-generator/main.c

```diff
@@ -52,8 +202,7 @@ static int ensure_root_fs_shared(const char *normal_dir)
 		fprintf(stderr, "cannot open or parse /proc/1/mountinfo\n");
 		return 1;
 	}
-
-	sc_mountinfo_entry *root = find_root_mountinfo(mounts);
+	sc_mountinfo_entry *root = find_dir_mountinfo(mounts, "/");
```

Comment:
This can return `NULL`. It's an existing problem but I would add a trivial check with a return.